docs: add a notice prohibiting retention policies on backup targets (#1010)
longhorn/website: content/docs/1.7.3/snapshots-and-backups/backup-and-restore/set-backup-target.md

Longhorn also supports setting up recurring snapshot/backup jobs for volumes, via Longhorn UI or Kubernetes Storage Class. See [here](../../scheduling-backups-and-snapshots) for details.

> **Notice**
> 

> > - The lifecycle of Longhorn backups within the backup store is entirely managed by Longhorn. **Any retention policy directly on the backup store is strictly prohibited**.

Reviewer: ```suggestion
> - The lifecycle of Longhorn backups within the backup store is managed by Longhorn. **Any retention policy on the backup store is prohibited**.
```

>

> > - Longhorn will attempt to clean up the backup-related custom resources in the following scenarios:

Reviewer: ```suggestion
> - Longhorn attempts to clean up the backup-related custom resources in the following scenarios:
```

> >   - An empty response from the NFS server due to a brief server downtime.

Reviewer: ```suggestion
>   - An empty response from the NFS server due to server downtime.
```

>   - A race condition between related Longhorn backup controllers.
> 

> >   The backup information will be resynchronized during the next polling interval. For more information, see [#9530](https://github.com/longhorn/longhorn/issues/9530).

Reviewer: ```suggestion
>   The backup information is resynchronized during the next polling interval. For more information, see [#9530](https://github.com/longhorn/longhorn/issues/9530).
```
